mcp/src/databricks_mcp/oauth_provider.py

```diff
@@ -0,0 +1,26 @@
+from databricks.sdk import WorkspaceClient
+from mcp.client.auth import OAuthClientProvider, TokenStorage
+from mcp.shared.auth import OAuthToken
+
+
+class DatabricksTokenStorage(TokenStorage):
+    def __init__(self, workspace_client):
+        self.workspace_client = workspace_client
+
+    async def get_tokens(self) -> OAuthToken | None:
+        headers = self.workspace_client.config.authenticate()
+        token = headers["Authorization"].split("Bearer ")[1]
```

Comment:
Is this format guaranteed? Maybe worth checking/throwing a better error if it's not